src/main/om/next.cljs

```diff
@@ -1827,12 +1827,13 @@
                              (not= c root)
                              props-change?)
                     (let [update-path (path c)]
-                      (loop [p (parent c)]
-                        (when (some? p)
-                          (let [update-path' (subvec update-path (count (path p)))]
-                            (update-props! p (assoc-in (props p) update-path' next-raw-props))
-                            (merge-pending-props! p)
-                            (recur (parent p)))))))))))))))
+                      (when-not (nil? update-path)
```

Comment:
just a tiny detail, I think the 2 lines above could be reduced to `(if-let [update-path (path c)]`